osmoutils/accum/accum.go

```diff
@@ -169,17 +169,21 @@ func (accum AccumulatorObject) GetPositionSize(addr sdk.AccAddress) (sdk.Dec, er
 // unclaimed rewards. The position's accumulator is also set to the current accumulator value.
 // Returns error if no position exists for the given address. Returns error if any
 // database errors occur.
-func (accum AccumulatorObject) ClaimRewards(addr sdk.AccAddress) (sdk.DecCoins, error) {
+func (accum AccumulatorObject) ClaimRewards(addr sdk.AccAddress) (sdk.Coins, error) {
 	position, err := getPosition(accum, addr)
 	if err != nil {
-		return sdk.DecCoins{}, NoPositionError{addr}
+		return sdk.Coins{}, NoPositionError{addr}
 	}
 
 	totalRewards := getTotalRewards(accum, position)
 
+	// Return the integer coins to the user
+	// The remaining change is reinvested into the new position.
+	truncatedRewards, remainingChange := totalRewards.TruncateDecimal()
+
 	// Create a completely new position, with no rewards
 	// TODO: remove the position from state entirely if numShares = zero
-	createNewPosition(accum, addr, position.NumShares, sdk.NewDecCoins(), position.Options)
+	createNewPosition(accum, addr, position.NumShares, remainingChange, position.Options)
```

Comment:
Are you confident leaving `remainingChange` in the position is necessary here? My understanding is that this can at most ever  be a rounding error and feels like it would be prudent to just shave it off for security reasons (the most harm that can be done is 1 token i.e. `1 uosmo`, which is error we accept in many places in GAMM)

Reply:
Im leaning towards Alpin's point on this one. I especially think of a world in which there is a bunch of accounts that claim rewards that never intend on creating a new position, so we just have state bloat that is worth nothing to the user.

Reply:
I was questioning this as well. Going to change to throw away this DecCoins change

Reply:
addressed